File: app/src/main/java/com/adsamcik/signalcollector/enums/ChallengeDifficulty.java
package com.adsamcik.signalcollector.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static com.adsamcik.signalcollector.enums.ChallengeDifficulty.EASY;
import static com.adsamcik.signalcollector.enums.ChallengeDifficulty.HARD;
import static com.adsamcik.signalcollector.enums.ChallengeDifficulty.MEDIUM;
import static com.adsamcik.signalcollector.enums.ChallengeDifficulty.VERY_EASY;
import static com.adsamcik.signalcollector.enums.ChallengeDifficulty.VERY_HARD;

@IntDef({VERY_EASY, EASY, MEDIUM, HARD, VERY_HARD})
@Retention(RetentionPolicy.SOURCE)
public @interface ChallengeDifficulty {
	int VERY_EASY = 0;
	int EASY = 1;
	int MEDIUM = 2;
	int HARD = 3;
	int VERY_HARD = 4;
}

File: app/src/main/java/com/adsamcik/signalcollector/enums/ChallengeDifficulty.kt
package com.adsamcik.signalcollector.enums

import android.support.annotation.IntDef

object ChallengeDifficulties {
    const val UNKNOWN = -1
    const val VERY_EASY = 0
    const val EASY = 1
    const val MEDIUM = 2
    const val HARD = 3
    const val VERY_HARD = 4

    @IntDef(UNKNOWN, VERY_EASY, EASY, MEDIUM, HARD, VERY_HARD)
    @Retention(AnnotationRetention.SOURCE)
    annotation class ChallengeDifficulty
}
